modules/internal/DefaultRectangular.chpl

```diff
@@ -2150,15 +2150,15 @@ module DefaultRectangular {
       writeln("Whose chunks are: ", rngs);
     }
 
-    var state: [1..numTasks] resType;
+    var state: [0..#numTasks] resType;
```

Comment:
Even though you said it in the merge message, I never remember that chunks is an iterator. I wonder if it'd be helpful to use `rngs.domain` for the array declarations and loop iterations. I'm fine sticking with this, but your previous PRs got me liking index neutral code.

Reply:
I like that idea.  It makes me sad to think how much cleaner all this code I've checked in this week would be if I'd had the ideas to implement the index-neutral features prior to these changes...  I'm going to use rngs.indices rather than rngs.domain, though (and will change the occurrences of .low to rngs.indices.low).

Reply:
Addressed in https://github.com/chapel-lang/chapel/pull/14800/commits/98f255a83be713c8418e350551910f4cc0e5769e